bin/improver-temporal-interpolate

```diff
@@ -0,0 +1,121 @@
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Script to interpolate data between validity times"""
+
+import warnings
+
+from improver.argparser import ArgParser
+from improver.utilities.load import load_cube
+from improver.utilities.save import save_netcdf
+from improver.utilities.temporal import (
+    cycletime_to_datetime, TemporalInterpolation)
+
+
+def main():
+    """
+    Interpolate data to intermediate times between the validity times of two
+    cubes. This can be used to fill in missing data (e.g. for radar fields) or
+    to ensure data is available at the required intervals when model data is
+    not available at these times.
+    """
+    parser = ArgParser(
+        description='Interpolate data between validity times ')
+
+    parser.add_argument('data_t0', metavar='DATA_T0',
+                        help='Cube contain the data at the beginning'
+                        ' of the period')
+
+    parser.add_argument('data_t1', metavar='DATA_T1',
+                        help='Cube contain the data at the end'
+                        ' of the period')
+
+    group = parser.add_mutually_exclusive_group(required=True)
+
+    group.add_argument("--interval_in_mins", metavar="INTERVAL_IN_MINS",
+                       default=None, type=int,
+                       help="Specifies the interval in minutes"
+                       " at which to interpolate "
+                       "between the two input cubes."
+                       " A number of minutes which does not "
+                       "divide up the interval equally will "
+                       "raise an exception. If intervals_in_mins "
```

Comment:
Could we have a CLI test for this exception please.

Reply:
No there is already a unit test for this.

Reply:
I'm sorry but that's not my point.  What happens when somebody changes the plugin and its unit tests?  We have a comment here that says the CLI will behave in a specific way - presumably we need it to continue to behave in that way.  So we need a test for that please.

Reply:
I will take out the comment if you like but I am not adding a CLI test. Sorry.

Reply:
As the exception is at the plugin level and is covered by a test we will forego a further test at the CLI level. We will leave the comment in as it is useful information to the user.

Reply:
For the record, I am not happy with this.  If it is important for the CLI to behave in this way, then that behaviour needs to be covered by a CLI test.  The point of testing at all levels is to be able to identify wider impacts of any changes in the guts of the code, as we would not expect to have to be always aware of all the places we are calling a plugin or utility.  For the sake of 5 minutes of work, we are setting ourselves up for confusion and problems down the line.